frontend/src/router/index.ts

```diff
@@ -117,6 +126,21 @@ router.beforeEach((to, from, next) => {
       next();
       return;
     }
+
+    if (to.name === "provider") {
+      const provider = to.params.provider as string;
+      if (!provider || !isValidCloudProvider([provider])) {
+        next({ name: "404" });
```

Comment:
we can just redirect to home

Reply:
If we just redirect to home, the user will not sense the error in the URL.
So maybe redirecting to a 404 page and then letting the user manually click `back home` is a better choice.